homeassistant/components/simplisafe/camera.py

```diff
@@ -0,0 +1,151 @@
+import asyncio
+
+from haffmpeg.camera import CameraMjpeg
+from haffmpeg.tools import IMAGE_JPEG, ImageFrame
+from simplipy.system import SystemStates
+
+from homeassistant.components.camera import Camera
+from homeassistant.components.ffmpeg import DATA_FFMPEG
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_aiohttp_proxy_stream
+
+from . import SimpliSafeEntity
+from .const import DATA_CLIENT, DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up SimpliCam cameras based on a config entry."""
+    simplisafe = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
+
+    cameras = []
+    for system in simplisafe.systems.values():
+        if (
+            "cameras" in system._location_info["system"]
+            and len(system._location_info["system"]["cameras"]) > 0
+        ):
+            for cam in system._location_info["system"]["cameras"]:
+                cameras.append(cam)
+
+    async_add_entities(
+        [
+            SimpliCam(simplisafe, system, hass.data[DATA_FFMPEG], camera)
+            for camera in cameras
+        ]
+    )
+
+
+class SimpliCam(SimpliSafeEntity, Camera):
+    """An implementation of a SimpliCam camera."""
+
+    def __init__(self, simplisafe, system, ffmpeg, camera):
+        """Initialize a SimpliCam camera."""
+        Camera.__init__(self)
+        super().__init__(
+            simplisafe,
+            system,
+            camera["cameraSettings"]["cameraName"],
+            serial=camera["uuid"],
+        )
+
+        self._simplisafe = simplisafe
+        self._system = system
+        self._camera = camera
+        self._ffmpeg = ffmpeg
+        self._last_image = None
+
+        self._is_online = self._camera["status"] == "online"
+        self._is_subscribed = self._camera["subscription"]["enabled"]
+
+    @property
+    def unique_id(self):
+        """Return unique ID of camera."""
+        return "{}-camera".format(self._camera["uuid"])
+
+    @property
+    def device_info(self):
+        """Return device registry information for this entity."""
+        return {
+            "identifiers": {(DOMAIN, self._camera["uuid"])},
+            "manufacturer": "SimpliSafe",
+            "model": self._camera["model"],
+            "name": self._camera["cameraSettings"]["cameraName"],
+            "via_device": (DOMAIN, self._system.serial),
+        }
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self._is_online and self._is_subscribed
+
+    @callback
+    def async_update_from_rest_api(self):
+        """Update the entity with the provided REST API data."""
+        self._is_online = self._camera["status"] == "online"
+        self._is_subscribed = self._camera["subscription"]["enabled"]
+
+    @property
+    def _video_url(self):
+        """Provide the video URL."""
+        return '{} -i "https://media.simplisafe.com/v1/{}/flv?x=1280&audioEncoding=AAC"'.format(
```

Comment:
This shouldn't be so explicit and should be abstracted away in `simplisafe-python`. In addition, please see [this ongoing `simplisafe-python` issue re: camera support](https://github.com/bachya/simplisafe-python/issues/85); the URL you use here isn't long-lived and has several other issues.

Reply:
The URL is long lived, the auth token isn't. It works for the odd check of the camera. Yes, if you want to continuously stream for hours, then this is not a good implementation, but for the odd check for a few minutes it's absolutely fine.

Reply:
That's good to know. Nevertheless, HASS shouldn't be aware of these implementation details; they belong in [`simplisafe-python`](https://github.com/bachya/simplisafe-python).